src/main/java/com/hotels/mutantswarm/MutantSwarmRunner.java

```diff
@@ -59,7 +60,7 @@ public void run(final RunNotifier notifier) {
       testNotifier.addFailure(t);
     } finally {
       try {
-        SwarmResults swarmResults = MutantSwarmRule.getSwarmResults();
+        SwarmResults swarmResults = core.getSwarmResults(MutantSwarmRule.getContextRef().get());
```

Comment:
This runner is still coupled to the rule so I don't think it needs to use the core? It can still use the rule and then getSwarmResults() on the rule can use its own contextRef so you don't need to have a getter for it and expose it externally. If the runner is supposed to be generic across the rule and extension then what's happening here now doesn't look correct.

Reply:
I am not sure what you mean; the method `getSwarmResults()` is stored in the core so we need to refer to the core to access it right?

Reply:
I'll show you in a PR or commit from me later, we can discuss tomorrow morning.